src/coll/mod.rs

```diff
@@ -104,13 +104,16 @@ impl Collection {
         };
 
         let mut read_preference = self.read_preference.clone();
+        let mut batch_size = 1;
```

Comment:
I'd just make this an option initially set to `None` and then changed to `Some(batch_size)` on line 115.